core/src/main/java/org/polypheny/db/adapter/AdapterManager.java

```diff
@@ -123,20 +133,61 @@ public DataSource getSource( int id ) {
             throw new RuntimeException( "Unknown adapter type: " + adapterType );
         }
         List<AdapterInformation> result = new LinkedList<>();
-        try {
-            //noinspection unchecked
-            for ( Class<DataStore> clazz : classes ) {
-                // Exclude abstract classes
-                if ( !Modifier.isAbstract( clazz.getModifiers() ) ) {
-                    String name = (String) clazz.getDeclaredField( "ADAPTER_NAME" ).get( null );
-                    String description = (String) clazz.getDeclaredField( "DESCRIPTION" ).get( null );
-                    List<AdapterSetting> settings = (List<AdapterSetting>) clazz.getDeclaredField( "AVAILABLE_SETTINGS" ).get( null );
-                    result.add( new AdapterInformation( name, description, clazz, settings ) );
+
+        //noinspection unchecked
+        for ( Class<DataStore> clazz : classes ) {
+            // Exclude abstract classes
+            if ( !Modifier.isAbstract( clazz.getModifiers() ) ) {
+                Map<String, List<AbstractAdapterSetting>> settings = new HashMap<>();
+
+                AdapterProperties properties = clazz.getAnnotation( AdapterProperties.class );
+                if ( properties == null ) {
+                    throw new RuntimeException( clazz.getSimpleName() + " does not annotate the adapter correctly" );
+                }
+
+                // used to evaluate which mode is used when deploying the adapter
+                settings.put( "mode",
+                        Collections.singletonList(
+                                new AbstractAdapterSettingList(
+                                        "mode",
```

Comment:
Vielleicht solche Dinge in constanten auslagern?

Reply:
I think in this case this would add more confusion, as those two strings are not necessarly the same and for 2-3 cases this seems kind of a overkill for me. What do you think?